lib/oban/query.ex

```diff
@@ -101,14 +101,18 @@ defmodule Oban.Query do
 
     with {:ok, query, lock_key} <- unique_query(changeset),
          :ok <- acquire_lock(conf, lock_key, query_opts),
-         {:ok, job} <- unprepared_one(conf, query, query_opts) do
-      return_or_replace(conf, query_opts, job, changeset)
+         {:ok, job} <- unprepared_one(conf, query, query_opts),
+         {:ok, job} <- return_or_replace(conf, query_opts, job, changeset) do
+      {:ok, Map.put(job, :duplicate?, true)}
```

Comment:
Since it is a virtual field we know it will be present. Let's use the struct update syntax instead of Map.put here.